libbeat/common/field.go

```diff
@@ -313,3 +313,22 @@ func ConcatFields(a, b Fields) (Fields, error) {
 	fields := make(Fields, 0, len(a)+len(b))
 	return append(append(fields, a...), b...), nil
 }
+
+// NewFieldsFromYAML reads the given YAML and parses it into a
+// Fields structure
+func NewFieldsFromYAML(data []byte) (Fields, error) {
+	var keys []Field
+
+	cfg, err := yaml.NewConfig(data)
+	if err != nil {
+		return nil, err
+	}
+	cfg.Unpack(&keys)
```

Comment:
error check?

Reply:
added a bunch of error checks now: https://github.com/elastic/beats/pull/10963/commits/8b03d7026cb52fe46f559946fa92492d7294720a. Not just on the function that was added in this PR. However, I wonder if this means we've been silently ignoring invalid configs and that this might brake a bunch of peoples installations.